### PR TITLE
Added a new boolean function - equals and int hashCode

### PR DESCRIPTION
**Pull request by 4MoPDai**
There was used Class Object and construction instanceof for different exceptions. And was created hashCode

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +27,29 @@ public Integer getLength() {
...
+                return false;
+            }
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        return 31 * length - 13 + 2 * this.width;
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Rectangle.java]: @4MoPDai why you are using 31, 13 and 2 in your solution?
this is just intersting for me. Why not for example 42 24 and 5?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +27,29 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj instanceof Rectangle) {
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Rectangle.java:36]: @4MoPDai please use color as well as length and width

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Rectangle.java (hunk 1 above)
Comment (4MoPDai) [reply src/main/java/core/basesyntax/Rectangle.java]: > @4MoPDai why you are using 31, 13 and 2 in your solution?
> this is just intersting for me. Why not for example 42 24 and 5?
        // Я использую 31, 13, 2. Потому что это уменьшает колизи.
        // то есть уменьшает возможность похожего хэш-кода у других
        //объектов

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -27,4 +27,37 @@ public Integer getLength() {
...
+            Rectangle rectObj = (Rectangle) obj;
+
+            if (!this.length.equals(rectObj.length)) {
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Rectangle.java]: @4MoPDai please simplify your if statements

**Review by boroda4436 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**